src/firmware/shared/physics_linear_algebra.h

```diff
@@ -9,8 +9,9 @@
  * \param[in] rhs_len the size of the input vector
```

Comment:
looks like this file only has matrix operations, probably can just toss it in `matrix.h`?

so it becomes `shared_matrix_multiply()` and `shared_matrix_transpose()`?

Reply:
porting those functions to use the `Matrix_t` struct is out of scope

Reply:
https://github.com/UBC-Thunderbots/Software/issues/2111

Reply:
Probably at least worth filing a ticket. I've definitely noticed that there are lots of things (like `dot2D`) that should be moved to the math library.